Plugins/Public/playercntl_plugin/PimpShip.cpp

```diff
@@ -271,10 +279,60 @@ namespace PimpShip
 		if (!mapInfo[iClientID].bInPimpDealer || !set_bEnablePimpShip)
 			return false;
 
-		PrintUserCmdText(iClientID, L"Available items: %d", mapAvailableItems.size());
-		for (map<uint, ITEM_INFO>::iterator iter = mapAvailableItems.begin(); iter != mapAvailableItems.end(); iter++)
+		uint beginFrom = 1;
+		uint endAt = mapAvailableItems.size();
+
+		ushort index = 0;
+		for (wstring::const_iterator it = wscParam.begin(); it != wscParam.end(); it++)
+		{
+			index++;
+
+			if (*it == '-')
+			{
+				if (it == wscParam.begin() && it == wscParam.end() - 1)
+				{
+					beginFrom = 1;
+					endAt = mapInfo[iClientID].mapCurrEquip.size();
+				}
+				else if (it == wscParam.end() - 1)
+				{
+					beginFrom = ToUInt(wscParam.substr(0, index));
+					endAt = mapInfo[iClientID].mapCurrEquip.size();
+				}
+				else if (it == wscParam.begin())
+				{
+					beginFrom = 1;
+					endAt = ToUInt(wscParam.substr(index));
+
+					if (endAt < beginFrom)
+						swap(endAt, beginFrom);
+				}
+				else
+				{
+					beginFrom = ToUInt(wscParam.substr(0, index));
+					endAt = ToUInt(wscParam.substr(index));
+
+					if (endAt < beginFrom)
+						swap(endAt, beginFrom);
+				}
+
+				break;
+			}
+		}
+
+		if (endAt > mapAvailableItems.size())
+			endAt = mapAvailableItems.size();
+
+		if (beginFrom == 0)
+			beginFrom = 1;
+
+		if (beginFrom > endAt)
+			beginFrom = endAt + 1;
```

Comment:
this seems wrong? shouldn't we swap beginFrom and endAt in this case?

Reply:
No. All swapping is done before. It checks if beginning is not getting out of list bounds (more than 137).

Reply:
So this could be `if (beginFrom > endAt + 1)`, but ok